### PR TITLE
User Interface work

### LINKED ISSUE #44: ERA A+V homepage look and feel
Need to get time from Natasha when she is back to work on the homepage look and feel how we link ERA AV from ERA and we link ERA AV from the library website. Having a look and feel similar to ERA would be nice.

Some example Avalon websites
https://mdpi.iu.edu/
https://media.dlib.indiana.edu/
https://media.northwestern.edu/

### PR DESCRIPTION
**Pull request by nnunn**
Fixes #44, #120, #63, #18, #16, #11

Over haul of the user experience/interface.

### CODE REVIEW

File: app/views/catalog/_home.html.erb [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      </div>
+  </div>
+<div class="row">
+    <div class="col-sm-8 info-pane">
+     <p>Welcome to ERA Audio<span class="glyphicon glyphicon-plus" aria-hidden="true"></span>Video (ERA A<span class="glyphicon glyphicon-plus" aria-hidden="true"></span>V), the University of Alberta's media streaming repository. ERA A<span class="glyphicon glyphicon-plus" aria-hidden="true"></span>V is a new feature of the University of Alberta's <a href="https://era.library.ualberta.ca">Education and Research Archive (ERA)</a>. ERA A<span class="glyphicon glyphicon-plus" aria-hidden="true"></span>V provides access to audiovisual materials in support of teaching, learning, and research at the University of Alberta. ERA A<span class="glyphicon glyphicon-plus" aria-hidden="true"></span>V is developed and supported by the <a href="https://library.ualberta.ca/">University of Alberta Libraries</a>.</p>
+
+</div>
+  <div class="col-sm-4">
+    <a href="/catalog?f[collection_ssim][]=Nosotros+Television+Programme&q=&search_field=all_fields#" class="featured">
```
Comment (murny) [inline app/views/catalog/_home.html.erb]: should use a rails link_to helper instead?

File: public/code.txt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<link rel="apple-touch-icon-precomposed" sizes="57x57" href="apple-touch-icon-57x57.png" />
```
Comment (murny) [inline public/code.txt]: Never seen a `code.txt` before? What's this for?

**Comment by nnunn** [comment]
Thanks for catching all these. I'll work through them tomorrow. If I have any questions I'll let you know.

File: app/assets/stylesheets/avalon.css.scss [hunk 3]
```diff
@@ -39,12 +39,7 @@
   }
 }
 
-a:not(.btn) {
-  color: $linkColor;
-  &:hover { color: $linkColor; }
-  &:active { color: $linkColor; }
-  &:visited { color: $linkColor; }
-}
+
```
Comment (nnunn) [inline app/assets/stylesheets/avalon.css.scss]: This stylesheet was not created by me and has loads of extra spaces. I suppose I shouldn't have added to the mess, but it didn't seem to matter. I'll clean up the entire file.

Comment (nnunn) [inline app/assets/stylesheets/avalon.css.scss]: Actually looking closer at it I don't want to fix all the spacing issue in this one without talking to Chris first.

File: .gitignore [hunk 4]
```diff
@@ -82,3 +82,9 @@ cucumber-results/
 
 /dropbox
 public/source_maps
+
+## From Docker deploy
+Gemfile.local
+config/lti.yml
+config/resque.yml
```
Comment (nnunn) [inline .gitignore]: This was suggested to me by Chris.

File: app/views/catalog/_home.html.erb [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            </div>
+          <button type="submit" class="call-to-action">Search ERA A<span class="glyphicon glyphicon-plus" aria-hidden="true"></span>V</button>
+        </form>
+         <%= link_to 'Browse ERA A<span class="glyphicon glyphicon-plus" aria-hidden="true"></span>V'.html_safe, catalog_index_path(search_field: 'all_fields', utf8: '✓', q: ''), :class => "call-to-action", :id => "browse-call" %>
+      </div>
+      </div>
+      <div class="video-wrap">
+      <object class="embed-responsive embed-responsive-16by9 banner-video">
+
+      <script>
```
Comment (nnunn) [inline app/views/catalog/_home.html.erb]: Great I will use ERB.

File: app/views/catalog/_home.html.erb (hunk 1 above)
Comment (nnunn) [inline app/views/catalog/_home.html.erb]: hmmm I took this from what Chris had on his version -- I couldn't figure out how to create the link_to so I just copied from his code. https://github.com/ualbertalib/avalon/blob/master/app/views/catalog/_home_text.html.erb

File: app/views/layouts/home.html.erb [hunk 6]
```diff
@@ -0,0 +1,50 @@
+<%#
```
Comment (nnunn) [inline app/views/layouts/home.html.erb]: sorry this file was supposed to be deleted. It is now.

File: app/assets/javascripts/avalon.js [hunk 7]
```diff
@@ -114,3 +114,5 @@ function timeCodeToSeconds(hh_mm_ss){
       tc_in_seconds = ( tc_hh * 3600 ) + ( tc_mm * 60 ) + tc_ss;
   return tc_in_seconds;
 }
+
+
```
Comment (nnunn) [inline app/assets/javascripts/avalon.js]: done

File: app/views/_user_util_links.html.erb [hunk 8]
```diff
@@ -13,12 +13,15 @@ Unless required by applicable law or agreed to in writing, software distributed
...
+<div class="mast-branding hidden-xs">
+  <ul class="col-lg-8 col-lg-offset-2">
+  
     <% unless current_page?(root_url) %>
-    <li><%= link_to 'Home', root_path %></li>
+ 
+    <li><%= link_to 'home', root_path %></li>
     <% end %>
-    <li><%= link_to 'Browse', catalog_index_path(search_field: 'all_fields', utf8: '✓', q: '') %></li>
+    <li><%= link_to 'browse', catalog_index_path(search_field: 'all_fields', utf8: '✓', q: '') %></li>
```
Comment (nnunn) [inline app/views/_user_util_links.html.erb]: I just copied this from the existing code:

https://github.com/ualbertalib/avalon/blob/master/app/views/_user_util_links.html.erb

File: app/views/_user_util_links.html.erb [hunk 9]
```diff
@@ -47,5 +50,8 @@ Unless required by applicable law or agreed to in writing, software distributed
     <%# Fallback if the test above fails %>
     <%= link_to 'Sign in', new_user_session_path %>
     <% end %>
+     <li><%= link_to 'contact', contact_path %></li>
+        <li><%= link_to 'ERA home', "https://era.library.ualberta.ca" %></li>
```
Comment (nnunn) [inline app/views/_user_util_links.html.erb]: yup, fixed.

File: .gitignore (hunk 4 above)
Comment (nnunn) [inline .gitignore]: I will ask about it on Monday.

File: app/views/modules/_header.html.erb [hunk 10]
```diff
@@ -10,31 +10,51 @@ Unless required by applicable law or agreed to in writing, software distributed
...
+        <h2 class="hidden-xs">education &amp; research archive</h2>
+        <h1><a href="/">ERA</a></h1>
+        <h3><span class="glyphicon glyphicon-triangle-right" aria-hidden="true"></span>AUDIO <span class="glyphicon glyphicon-plus logo-plus" aria-hidden="true"></span> VIDEO</h3>
+      </div>
+      <div class="col-sm-4 user-util hidden-xs">
         <%= render partial: "modules/user_util_dropdowns"%>
       </div>
+     
+          <div class="col-xs-12 mobile-search visible-xs">
+ <form accept-charset="UTF-8" action="/catalog" id="search-form-header" method="get" role="search" class="form-inline">
```
Comment (nnunn) [inline app/views/modules/_header.html.erb]: Once again - this was copy and pasted from existing code (I believe it was written by someone at Indiana University). 

https://github.com/ualbertalib/avalon/blob/master/app/views/modules/_header.html.erb

I don't mean to be skiving off on this work, but I also don't really feel that it's my place to rewrite code. My sense was I was brought in to make the UI look good by hooking into existing code...

**Comment by nnunn** [comment]
Ok - I've gone through the requests and changed things that were mistakes/issues that I created. I am going to wait on other changes for Chris to get back. Specifically rewriting code to make it more elegant and standards compliant...I would like to check that reworking code is in my scope of work.

File: app/views/catalog/_home.html.erb (hunk 5 above)
Comment (nnunn) [inline app/views/catalog/_home.html.erb]: Done and much more elegant - thanks!

**Comment by weiweishi** [comment]
@cwant As @murny and @nnunn noted, there are some issues with the code we inherited from upstream. I've created https://github.com/ualbertalib/avalon/issues/166 and https://github.com/ualbertalib/avalon/issues/167. I will leave you to decide whether you'd like to address them in separate PRs, or add the fix here. 

Re: code fixes from upstream, let's evaluate the importance of these fixes. I think if they are important enough (like the proper rails helpers to address security issue), we should make the effort to at least fix them in the files we touch.

File: .gitignore (hunk 4 above)
Comment (cwant) [inline .gitignore]: The docker script comes primarily from upstream the avalon project, and modifies these files. I think they are safe to ignore here. Alternative would be to use the "git update-index --assume-unchanged" trick, but I think this change makes things more explicit.

**Comment by cwant** [comment]
@nnunn On wide screens, there is a misalignment in the two panels in the left sidebar:

unaligned-sidebar

Doesn't happen in narrower resolutions.

I can look into this if your time is tight.

**Comment by cwant** [comment]
The widget that holds the login stuff (pulldown) is too close to the top of the screen:

logged-in-thingy-too-close-to-top

(Doesn't match the vertical alignment of the U of A logo.)

On narrower resolutions, this widget gets partially cut off on the right side.

**Comment by cwant** [comment]
On non-landing page, there is no search box:

no-search

Probably need to put it back in the header and add it conditionally.

**Comment by nnunn** [comment]
So I can't login on the compuserve canada system so that's why I didn't see any of these issues. Is there a way I can see the login UI on that server?

**Comment by nnunn** [comment]
I'll take a look.

**Comment by cwant** [comment]
There are 9 test suite failures:

rspec ./spec/features/capybara_homepage_spec.rb:62 # checks navigation to external links verifies presence of features after login
rspec ./spec/features/capybara_navigation_spec.rb:18 # checks navigation after logging in checks navigation to Browse
rspec ./spec/features/capybara_navigation_spec.rb:32 # checks navigation after logging in checks navigation to Manage Content
rspec ./spec/features/capybara_navigation_spec.rb:47 # checks navigation after logging in checks naviagtion to Manage Groups
rspec ./spec/features/capybara_navigation_spec.rb:60 # checks navigation after logging in checks naviagtion to Playlist
rspec ./spec/features/capybara_navigation_spec.rb:69 # checks navigation after logging in is able to sign out
rspec ./spec/features/capybara_navigation_spec.rb:79 # Search is able to enter keyword and perform search
rspec ./spec/features/capybara_navigation_spec.rb:85 # Search gives appropriate error when keyword returns no results
rspec ./spec/features/capybara_playlist_spec.rb:18 # Playlist checks navigation when create new playlist is accessed

I can look into these -- most are related to text that the upstream project would like to see on certain web pages (e.g., "Powered by Avalon", or URL to avalonmediasystem.org, etc).

**Comment by cwant** [comment]
There is no nav menu on mobile version on any of the non-home pages:

no-nav-on-mobile

By the way Natasha, if there are any of these things I'm bringing up that you would prefer than I look at, don't hesitate to ask. I'm looking at the test suite issues now.

**Comment by nnunn** [comment]
Nope I'm good - I've made most of the fixed already. I'm just working on replacing the forms with form_for tags. I'll look into the mobile menu in a little bit.

**Comment by nnunn** [comment]
Ok I've fixed some of the issues - I want to talk a little about the mobile menu at the standup. I also found a bug that I think I'll need your help with.

**Comment by seanluyk** [comment]
Apologies for not catching this sooner, @nnunn. I have a few textual changes to request, which I'll put into a Google doc and send to you

**Comment by nnunn** [comment]
@seanluyk - I'm not sure comment edits get sent to you. I edited the comment above with a question for you...

File: app/views/layouts/avalon.html.erb [hunk 11]
```diff
@@ -18,26 +18,49 @@ Unless required by applicable law or agreed to in writing, software distributed
...
     <a id="skip-to-content" class="sr-only sr-only-focusable" href="#content">Skip to main content</a>
     <%= render partial: 'shared/ajax_modal' %>
-    <%= render 'modules/header' %>
-    <div class="container-fluid">
-      <%= render :partial => 'modules/global_navigation' %>
+    
+    <% if !current_page?(root_path) || request.query_string.present? %>
+        <%= render 'modules/header' %>
+        <%= render :partial => 'modules/global_navigation' %>
+        <div class="container-fluid secondary">
```
Comment (cwant) [inline app/views/layouts/avalon.html.erb]: Hey Natasha, does this div get closed? (if it does, it's not in this file). I wonder if this should be outside of the if block?

Comment (cwant) [inline app/views/layouts/avalon.html.erb]: By the way, regarding the header problem we discussed, do you think this change will work for you?

https://github.com/ualbertalib/avalon/commit/f676de11b42198168787a3d52729efb735820d8b

File: app/views/layouts/avalon.html.erb [hunk 12]
```diff
@@ -53,13 +53,14 @@ Unless required by applicable law or agreed to in writing, software distributed
...
     
-    <% if !current_page?(root_path) || request.query_string.present? %>
+    <% if !current_page?(root_path) %>
         <%= render 'modules/header' %>
         <%= render :partial => 'modules/global_navigation' %>
         <div class="container-fluid secondary">
     <% end %>  
       <%= render 'modules/flash_messages' %>
         <%= yield %>
+        </div>
```
Comment (cwant) [inline app/views/layouts/avalon.html.erb]: Okay, new problem now: if the current page is the root path, this closing tag for the div doesn't have a matching opening tag.

Comment (nnunn) [inline app/views/layouts/avalon.html.erb]: right - I may need to use the same logic to hide and show the closing div tag.

But... your fix for the catalog bug is close, but there are situation where there is no query param eg /catalog?commit=Search+ERA+A%2BV&utf8=✓ -- in this case the double header appears.

**Comment by nnunn** [comment]
Ok - I think all the issues are fixed here. I am still waiting for @seanluyk to send me his text changes and weigh in on the mobile menu issue. 

I have replace the <form> tags with  <%= form_tag

**Comment by seanluyk** [comment]
@cwant @nnunn  - on the mobile issue - non-admin/manager users will likely only see home, browse, playlists, and sign out, so I think this can be hidden from mobile view from now. This may change later if we see a need (e.g. lots of use of playlists)

**Comment by seanluyk** [comment]
@nnunn text changes have been incorporated into this document: https://docs.google.com/a/ualberta.ca/document/d/1kbWbvCR5kGRiuIeBmSSTLsJaxST_v9bcJP1vPePQbDI/edit?usp=sharing

Please let me know if you have any questions!

**Comment by nnunn** [comment]
Ok - I think this PR is ready for approval.

**Comment by cwant** [comment]
Hi Natasha,

I still get clipping of the username on medium screen resolutions:

username-clip-home-page

Seems to only happen on home page.

**Comment by seanluyk** [comment]
More on the issue of spacing, I'm thinking that we should decide on one. and stick to it. I've personally gone back and forth between Audio + Video and Audio+Video and ERA A + V and ERA A+V. The latter option is easier to type so that's my personal preference. It seems cleaner as well. What do you think @cwant @umarqasim @nnunn?

**Comment by murny** [comment]
On this topic of your brand, you guys should also consider how your brand is being used across the website. Especially with regards to SEO and on other devices.  Many places are doing various things with your brand of "ERA A+V" or "ERA A + V" (and the many variations like "ERA AUDIO +
VIDEO"). Should probably choose one and remain consistent with it everywhere.

Also think about implementations of this. Such as how the brand is marked up in the header, only the ERA part is in the main heading (`h1`) while the rest of the brand is in sub headers (`h3`). (Note: looks like there is multiple H1s on a single page as well, which typically goes against SEO best practices.) 

Furthermore, a glyphicon (icon font) is being used for the plus sign (+) in a few places instead of a regular plus sign (`&plus;` or `+`). This could be concerning depending on how this plays out in SEO (google and others will probably ignore this icon font which changes your brand to ERA AV on search results, etc.). This also effects how other devices will view this. As the plus sign is currently being hidden on screen readers and the like ( since your using `aria-hidden="true"`). 

So in terms of branding, consistency is key. Probably shouldn't implement your brand in various ways across your website. SEO should also be an important consideration when building a website too.

**Comment by nnunn** [comment]
My vote is to leave it as ERA Audio+Video in the header and then call it ERA A+V in the body of the site. I know this isn't completely consistent but I think it makes sense (people will understand the shortened form). @murny thanks for the SEO reminders. I will remove all the glyphicons and make sure there is only one H1 tag. @cwant I will also fix the user name being cut off. I hope to have these changes pushed today - however I'll need a final verdict on the branding issue.

**Comment by umarqasim** [comment]
my vote is also for ERA A+V

**Comment by seanluyk** [comment]
Thanks, @murny for your great points on this issue! @nnunn I think the consensus is to go with ERA Audio+Video in the header and ERA A+V in the body as you suggest.  Please let me know if there are any remaining questions re: branding and I'll respond a.s.a.p.

**Comment by nnunn** [comment]
Ok those changes are pushed (along with the issue @cwant pointed out).

**Comment by umarqasim** [comment]
The footer has "About ERA Audio+Video" text, may be we should replace it with "About ERA A+V"

**Comment by nnunn** [comment]
Thanks for catching this - it is fixed. Also I am working on some html validation issues.

**Comment by nnunn** [comment]
@cwant I think I've fixed all the validation issues. If you have a moment to take a look that would be great (validators can be different so if yours still sees issues let me know). I've also made the text changes that @seanluyk  and @umarqasim requested. Please let me know if there are any other bugs/issues you come across.

**Comment by seanluyk** [comment]
@nnunn - not crucial for launch, but hoping it's a simple fix:

When logged into dev as any role above a regular end user, the publish/unpublish, edit, and delete buttons are far off to the left. Here's a screen shot:

image

When you have a chance, these should be brought in to the centre.

**Comment by seanluyk** [comment]
@nnunn as above, same issue on the edit screen (again, not crucial at this point as there are a limited number of users who can see this):

image

**Comment by seanluyk** [comment]
@nnunn Same issue on the pages in this menu (playlists looks fine):
image

**Comment by nnunn** [comment]
Thanks @seanluyk  for catching those layout issues. They are fixed in dev now.

**Comment by seanluyk** [comment]
Many thanks @nnunn. Everything looks great to me in terms of UI, unless @umarqasim or @cwant have any other comments

**Comment by umarqasim** [comment]
just poked around the site, the UI looks good to me

**Comment by nnunn** [comment]
Hopefully you mean remove underscores in css. Dashes are much better in my opinion.

**Comment by cwant** [comment]
@nnunn Yep, that's exactly what I meant :smiley_cat:

**Comment by nnunn** [comment]
👍

File: app/views/modules/_header.html.erb [hunk 13]
```diff
@@ -10,31 +10,57 @@ Unless required by applicable law or agreed to in writing, software distributed
...
-      </form>
-      <div class="col-md-12 user-util hidden-xs">
-        <%= render partial: "modules/user_util_dropdowns"%>
+    <div class="row">
+      <div class="lib-logo col-sm-4 hidden-xs">
+        <%= ualib_logo_link %>
+        <%= link_to "ERA Home", "https://era.library.ualberta.ca", :class => "era-link" %>
       </div>
+        <div class="mast-branding col-sm-4">
+          <h2 class="hidden-xs">education &amp; research archive</h2>
```
Comment (nnunn) [inline app/views/modules/_header.html.erb]: Yes I think it should be hidden.

Comment (nnunn) [inline app/views/modules/_header.html.erb]: wait sorry - I'm wrong. It should not be hidden. I've fixed this in the favicon PR.

**Pull request merged**